src/main/java/com/semmle/jira/addon/workflow/ResolutionValidatorFactory.java

```diff
@@ -0,0 +1,82 @@
+package com.semmle.jira.addon.workflow;
+
+import com.atlassian.jira.component.ComponentAccessor;
+import com.atlassian.jira.config.ResolutionManager;
+import com.atlassian.jira.issue.resolution.Resolution;
+import com.atlassian.jira.plugin.workflow.AbstractWorkflowPluginFactory;
+import com.atlassian.jira.plugin.workflow.WorkflowPluginConditionFactory;
+import com.atlassian.jira.plugin.workflow.WorkflowPluginValidatorFactory;
+import com.opensymphony.workflow.loader.AbstractDescriptor;
+import com.opensymphony.workflow.loader.ConditionDescriptor;
+import com.opensymphony.workflow.loader.ValidatorDescriptor;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ResolutionValidatorFactory extends AbstractWorkflowPluginFactory
```

Comment:
Make clear that the factory is shared by changing the name to `CheckResolutionFunctionFactory`